sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+
+      if (update_device_temp((int32_t)desired_temp))
+      {
+        send_reported_temperature_property(device_max_temp, -1);
+      }
+      break;
+
+    // A response from a twin reported properties publish message. With a successfull update of
+    // the reported properties, the payload will be empty and the status will be 204.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      printf("A twin reported properties response message was received\n");
+      break;
+  }
+}
+
+// Invoke the requested command if supported and return status | Return error otherwise
+static void handle_command_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_method_request* command_request)
+{
+
+  if (az_span_is_content_equal(report_command_name_span, command_request->name))
+  {
+    az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+    az_span command_payload_span
+        = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+    // Invoke command
+    uint16_t return_code;
+    az_result response = invoke_getMaxMinReport(
+        command_payload_span, command_response_span, &command_response_span);
+    if (response == AZ_ERROR_ITEM_NOT_FOUND)
+    {
+      return_code = 404;
+    }
+    else
+    {
+      return_code = 200;
+    }
+
+    // Send command response with report as JSON payload
+    int rc;
+    if ((rc = send_command_response(command_request, return_code, command_response_span)) != 0)
+    {
+      printf("Unable to send %u response, status %d\n", return_code, rc);
+    }
+  }
+  else
+  {
+    // Unsupported command
+    printf(
+        "Unsupported command received: %.*s.\n",
+        az_span_size(command_request->name),
+        az_span_ptr(command_request->name));
+
+    int rc;
+    if ((rc = send_command_response(command_request, 501, report_error_payload)) != 0)
+    {
+      printf("Unable to send %d response, status %d\n", 501, rc);
+    }
+  }
+}
+
+// Callback for incoming MQTT messages
+static int on_received(char* topicName, int topicLen, MQTTClient_message* message)
+{
+  if (topicLen == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
+  }
+
+  printf("Topic: %s\n", topicName);
+
+  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
+
+  // Parse the incoming message topic and check which feature it is for
+  az_iot_hub_client_twin_response twin_response;
+  az_iot_hub_client_method_request command_request;
+
+  if (az_succeeded(
+          az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response)))
+  {
+    printf("Twin Message Arrived\n");
+
+    // Determine what kind of twin message it is and take appropriate actions
+    handle_twin_message(message, &twin_response);
+
+    printf("Response status was %d\n", twin_response.status);
```

Comment:
>printf [](start = 4, length = 6)

Nit: this printf() should come above the handle_twin_message(), and it can be combined into the top printf("Twin Message arrived, response status was %d\n",...).  As it is, it appears the status was set by the handle_twin_message() call and we're outputing in response to that.

Reply:
And maybe less of a nit, but should we actually be checking that status and doing anything based on it??

---
In reply to: [452354398](https://github.com/Azure/azure-sdk-for-c/pull/862#discussion_r452354398) [](ancestors = 452354398)

Reply:
I can make it simple and only "handle" if the status is valid. Otherwise I can just print the status and continue